### ISSUE TITLE
Fix `overrides.files` negated pattern regression

### ISSUE THREAD

**Issue by emmasax** [stylelint/stylelint#7056]
### What minimal example or steps are needed to reproduce the bug?

Create file `fontSizeAllowedFile.tsx`:

```ts
import { css } from '@emotion/react';

export const someStyling = css`
  font-size: 16px;
  color: white;
`;
```

### What minimal configuration is needed to reproduce the bug?

Using glob pattern to exclude files (`!(files)`) does not work in Stylelint 15. It does work in previous versions.

Using the positive glob does include only this file `["*/**/fontSizeAllowedFile.tsx"]`, just negating it does not work.

```json
{
  "customSyntax": "postcss-styled-syntax",
  "rules": {
    "property-disallowed-list": null
  },
  "overrides": [
    {
      "files": ["!(*/**/fontSizeAllowedFile.tsx)"],
      "rules": {
        "property-disallowed-list": ["font-size", "font-weight"]
      }
    }
  ]
}
```

### How did you run Stylelint?

Via nx stylelint command

### Which Stylelint-related dependencies are you using?

```json 
{
    "postcss-styled-syntax": "^0.4.0",
    "stylelint": "^15.10.1",
    "stylelint-config-standard": "^33.0.0"
}
```

### What did you expect to happen?

File should not run disallowed list on the excluded file.

### What actually happened?

File is not excluded
```
 4:3  ✖  Unexpected property "font-size"    property-disallowed-list
```

**Comment by Mouvedia** [comment]
I guess it's just a matter of doing a `git bisect`, if it used to work.
The only PR I found which looks related is #​6547.

**Comment by jeddy3** [comment]
@emmasax Thanks for the detailed report and for using the template.

I'm able to reproduce this locally with:

```json
{
  "rules": {
    "color-no-invalid-hex": null
  },
  "overrides": [
    {
      "files": ["!(*/**/test.css)"],
      "rules": {
        "color-no-invalid-hex": true
      }
    }
  ]
}
```

```css
/* /test/test.css */
a {
  color: #00;
}
```

`15.10.1`:

```console:
jeddy3@mac stylelint-test % npx stylelint "**/*.css"

test/test.css
 2:10  ✖  Unexpected invalid hex color "#00"  color-no-invalid-hex

1 problem (1 error, 0 warnings)
```

`14.16.1`

```
jeddy3@mac stylelint-test % npx stylelint "**/*.css"
(no warnings)
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time. It may require a bit of digging, though.

**Comment by ybiquitous** [comment]
Sorry for the late response. As @Mouvedia guessed, I produced this bug through PR #6547. The bug is at:

https://github.com/stylelint/stylelint/blob/d881c9df941d22a8cdc52bbd13e1b1900b0f95f8/lib/augmentConfig.mjs#L394-L398

Here's a simple confirmation. Returning `true` by `isMatch()` for negation patterns (e.g.  `!(**/test.css)`) adds an input file path (e.g. `/foo/test.css`) to lint targets.

```js
micromatch = require('micromatch');
micromatch.isMatch('/foo/test.css', ['!(**/test.css)'], { dot: true, basename: true });
//=> true
```

I'll create a patch pull request soon.

However, this glob's syntax/behavior depends on a glob library (currently [`micromatch`](https://www.npmjs.com/package/micromatch) and is undocumented in the [Stylelint document](https://github.com/stylelint/stylelint/blob/d881c9df941d22a8cdc52bbd13e1b1900b0f95f8/docs/user-guide/configure.md#overrides). So, I recommend using [`ignoreFiles`](https://github.com/stylelint/stylelint/blob/d881c9df941d22a8cdc52bbd13e1b1900b0f95f8/docs/user-guide/configure.md#ignorefiles) instead for clarity and stability. E.g.

```diff
{
-	files: ['!(**/test.css)'],
+	files: ['**/*.css'],
+	ignoreFiles: ['**/test.css'],
}
